xml_converter/src/xml_converter.cpp

```diff
@@ -62,57 +62,69 @@ vector<string> get_files_by_suffix(string directory, string suffix) {
-                    continue;
-                }
-                move_supplementary_files(path, new_directory);
-            }
-            else if (has_suffix(filename, ".trl") || has_suffix(filename, ".xml")) {
-                continue;
-            }
-            else {
-                // TODO: Only include files that are referenced by the
-                // individual markers in order to avoid any unnessecary files
-                string new_path = join_file_paths(output_directory, filename);
-                copy_file(path, new_path);
-            }
+void read_taco_directory(
+    string input_path,
+    map<string, Category>* marker_categories,
+    vector<Parseable*>* parsed_pois) {
+    if (!filesystem::exists(input_path)) {
+        cout << "Error: " << input_path << " is not an existing directory or file" << endl;
+    }
+    else if (filesystem::is_directory(input_path)) {
+        vector<string> xml_files = get_files_by_suffix(input_path, ".xml");
+        for (const string& path : xml_files) {
+            parse_xml_file(path, input_path, marker_categories, parsed_pois);
         }
     }
+    else if (filesystem::is_regular_file(input_path)) {
+        parse_xml_file(input_path, get_base_dir(input_path), marker_categories, parsed_pois);
+    }
 }
 
-void read_taco_directory(string input_path, map<string, Category>* marker_categories, vector<Parseable*>* parsed_pois) {
+void read_burrito_directory(
+    string input_path,
+    map<string, Category>* marker_categories,
+    vector<Parseable*>* parsed_pois) {
     if (!filesystem::exists(input_path)) {
         cout << "Error: " << input_path << " is not an existing directory or file" << endl;
     }
     else if (filesystem::is_directory(input_path)) {
-        vector<string> xml_files = get_files_by_suffix(input_path, ".xml");
-        for (const string& path : xml_files) {
-            parse_xml_file(path, marker_categories, parsed_pois);
+        vector<string> burrito_files = get_files_by_suffix(input_path, ".bin");
+        for (const string& path : burrito_files) {
+            read_protobuf_file(path, input_path, marker_categories, parsed_pois);
         }
     }
     else if (filesystem::is_regular_file(input_path)) {
-        parse_xml_file(input_path, marker_categories, parsed_pois);
+        read_protobuf_file(input_path, get_base_dir(input_path), marker_categories, parsed_pois);
     }
 }
 
-void write_taco_directory(string output_path, map<string, Category>* marker_categories, vector<Parseable*>* parsed_pois) {
+void write_taco_directory(
+    string output_path,
+    map<string, Category>* marker_categories,
+    vector<Parseable*>* parsed_pois) {
     // TODO: Exportion of XML Marker Packs File Structure #111
     if (!filesystem::is_directory(output_path)) {
         if (!filesystem::create_directory(output_path)) {
             cout << "Error: " << output_path << "is not a valid directory path" << endl;
         }
+        return;
```

Comment:
I think this is in the wrong scope, it should be in the if statement above it for when a directory is not created properly. Otherwise this will always exit even if a directory was created properly.